### PR TITLE
User login and register

### PR DESCRIPTION
**Pull request by tzhao-ooc**
Register page: 

- error message should show properly for each field when user move to another field.
- register successfully should direct to log in page

Login page:
- error message should show for invalid username/password
- successful login should direct to dashboard page; showing notice message "login successfully" for 3 seconds

Nav bar:
- display login and register if no user logged in
- display profile and logout if user logged in

### CODE REVIEW

File: app/javascript/events.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-let cachedEvents = JSON.parse(localStorage.getItem("cachedEvents")) || [];
```
Comment (AndyZzzZzzZzz) [inline app/javascript/events.js:1]: Without the let keyword you are not creating and initializing a new variable. Does this work?

File: app/javascript/register.js [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  
+//         let valid = true;
+  
+//         // Get form fields
+//         const firstName = document.getElementById('user_first_name');
+//         const lastName = document.getElementById('user_last_name');
+//         const username = document.getElementById('user_username');
+//         const email = document.getElementById('user_email');
+//         const password = document.getElementById('user_password');
+//         const passwordConfirmation = document.getElementById('user_password_confirmation');
```
Comment (AndyZzzZzzZzz) [inline app/javascript/register.js:18]: Don't merge these comments into the main codebase

File: app/javascript/events.js (hunk 1 above)
Comment (tzhao-ooc) [reply app/javascript/events.js:1]: this fixed the error "cachedEvents has already been define".

File: app/javascript/register.js (hunk 2 above)
Comment (tzhao-ooc) [reply app/javascript/register.js:18]: will fixed in next commit

File: app/javascript/events.js (hunk 1 above)
Comment (AndyZzzZzzZzz) [reply app/javascript/events.js:1]: Where is this error coming from? Also, does the code work when first initializing?

Comment (tzhao-ooc) [reply app/javascript/events.js:1]: if you inspect the webpage, and go to console. you'll see those. It works on my side. you can also test this in user_login branch.

**Pull request merged**